great_expectations/datasource/fluent/interfaces.py

```diff
@@ -1256,12 +1257,14 @@ def _create_validator(self, *, result_format: ResultFormatUnion) -> V1Validator:
         )
 
     @overload
-    def compute_metrics(self, metrics: Metric) -> MetricResult: ...
+    def compute_metrics(self, metrics: Metric[_MetricResultT]) -> _MetricResultT: ...
 
     @overload
     def compute_metrics(self, metrics: list[Metric]) -> list[MetricResult]: ...
```

Comment:
AFAIK python type hints are unable to represent this kind of thing for heterogenous lists, so not touching the list version.

Reply:
Yeah I don't know how that would work